### PR TITLE
docs: add vue router tips

### PR DESCRIPTION
**Pull request by screetBloom**
##### Checklist

- [ ] `npm test` passes
- [ ] tests are included
- [x] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change
add a FAQ about Vue Router

<br>
Some background information：     

- I'm having some problems when  use vue router in qiankun. I hope I can share some of my experience with others.
- And this is my first contribution to qiankun,   **I am eager to get this merged** . Thx~ 

@kuitos
<br>
<br>

### CODE REVIEW

**Comment by screetBloom** [comment]
Some checks were not successful !      
**But this looks like it's due to an external cause** 👇.      

image


And my local looks normal 👇.
image

**Comment by kuitos** [comment]
merge with master then the ci failure will gone

**Comment by screetBloom** [comment]
> 补充一下英文文档部分？

好的，收到

**Comment by screetBloom** [comment]
我使用 sub-app 的原因：https://en.wiktionary.org/wiki/subapplication

根据wiki来看，应该是subapplication  或者 sub-app

@kuitos 你看一下是否认可，如果不认可的话我快速改一下

File: docs/faq/README.md [hunk 1]
```diff
@@ -71,6 +71,21 @@ To solve the error, choose one of the options listed below:
 1. Use bundler to pack `Vue` library, instead of CDN or external module
 2. Rename `Vue` to other name in master application, eg: `window.Vue2 = window.Vue; window.Vue = undefined`
 
+## Tips on Using Vue Router
+
+The qiankun main app activates the corresponding sub-app according to the `activeRule` configuration.
+
+### a. The main app is using Vue Router's hash mode
+
+When the main app is in hash mode, generally, sub-app is also in hash mode. In this case, the base hash path of the main app is assigned to the corresponding sub-app (e.g. `#base`). At this time, if the sub-app needs to make a secondary path jump in hash mode (such as `#/base1/child1`) on the basis of the base path, you just need to add a prefix  for each route yourself.
```
Comment (kuitos) [inline docs/faq/README.md:80]: 统一叫 micro app 就好了，中文文档这边统一是 微应用

Comment (kuitos) [inline docs/faq/README.md:80]: > make a secondary path jump in hash mode

应该是 `make a secondary path jump with hash mode` 吧？

> path jump in hash mode (such as `#/base1/child1`) on the basis of the base path

这段读起来怪怪的

> you just need to add a prefix  for each route yourself.

for 前面多了个空格

File: docs/faq/README.md [hunk 2]
```diff
@@ -71,6 +71,21 @@ To solve the error, choose one of the options listed below:
...
+The qiankun main app activates the corresponding sub-app according to the `activeRule` configuration.
+
+### a. The main app is using Vue Router's hash mode
+
+When the main app is in hash mode, generally, sub-app is also in hash mode. In this case, the base hash path of the main app is assigned to the corresponding sub-app (e.g. `#base`). At this time, if the sub-app needs to make a secondary path jump in hash mode (such as `#/base1/child1`) on the basis of the base path, you just need to add a prefix  for each route yourself.     
+The base parameter in VueRouter's hash mode [does not support adding a hash path base](https://github.com/vuejs/vue-router/blob/dev/src/index.js#L55-L69).
+
+### b. The main app is using Vue Router's history mode
+
+When the main app is in history mode and the sub-app is also in history mode, it performs perfectly. And if the sub-app needs to add a base path, just [set the base property](https://router.vuejs.org/api/#base) of the sub item.
```
Comment (kuitos) [inline docs/faq/README.md:85]: > it performs perfectly

it works perfectly 更贴切？

File: docs/faq/README.md [hunk 3]
```diff
@@ -71,6 +71,21 @@ To solve the error, choose one of the options listed below:
...
+
+When the main app is in history mode and the sub-app is in hash mode, it performs perfectly.
```
Comment (kuitos) [inline docs/faq/README.md:87]: 同上

File: docs/faq/README.md (hunk 1 above)
Comment (screetBloom) [reply docs/faq/README.md:80]: 【1】
我觉得这里还是用 `in hash mode` 比 `with hash mode` 比较好，我查了一下应该是用 `in hash mode` 的写法更普遍一些
- https://www.zhihu.com/question/267415597
- https://stackoverflow.com/questions/52821111/vuejs-router-in-hash-mode-route-to-error-page-if-file-not-found
- https://github.com/vuejs/vue-router/issues/1406

<br>
【2】

> At this time, if the sub-app needs to make a secondary path jump in hash mode (such as `#/base1/child1`) on the basis of the base path, you just need to add a prefix  for each route yourself. 

改成如下这样是不是更好：

At this time, if the sub-app needs to make a secondary path jump in hash mode (such as `#/base1/child1`) **when there is a base path**, you just need to add a prefix  for each route yourself.

Comment (screetBloom) [reply docs/faq/README.md:80]: 好的，收到
另外**非我本次提交**的范围需要改为 micro app 吗

Comment (kuitos) [reply docs/faq/README.md:80]: 可以单独再提一个 pr 做这个事情

Comment (kuitos) [reply docs/faq/README.md:80]: ok

**Pull request merged**